packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -526,16 +526,31 @@ export default function useAutocomplete(props) {
     // eslint-disable-next-line react-hooks/rules-of-hooks
     React.useEffect(() => {
       if (!inputRef.current || inputRef.current.nodeName !== 'INPUT') {
-        console.error(
-          [
-            `MUI: Unable to find the input element. It was resolved to ${inputRef.current} while an HTMLInputElement was expected.`,
-            `Instead, ${componentName} expects an input element.`,
-            '',
-            componentName === 'useAutocomplete'
-              ? 'Make sure you have binded getInputProps correctly and that the normal ref/effect resolutions order is guaranteed.'
-              : 'Make sure you have customized the input component correctly.',
-          ].join('\n'),
-        );
+        if (
+          inputRef.current &&
+          inputRef.current.nodeName === 'TEXTAREA' &&
+          componentName === 'Autocomplete'
+        ) {
+          console.warn(
+            [
+              `A ${inputRef.current} element was provided to ${componentName} where input was expected.`,
+              `This is not a supported scenario but it may work under certain conditions.`,
+              `A ${inputRef.current} keyboard navigation may conflict with Autocomplete controls (e.g. enter and arrow keys).`,
```

Comment:
`inputRef.current` is not a string, it's a DOM node. Since we know it's a textarea (as it's checked a few lines above), it's OK to just hardcode the name here.

```suggestion
              `A textarea element was provided to ${componentName} where input was expected.`,
              `This is not a supported scenario but it may work under certain conditions.`,
              `A textarea keyboard navigation may conflict with Autocomplete controls (e.g. enter and arrow keys).`,
```

Reply:
I noticed that it was a DOM node as well, but saw that it was used in this way in the initial error message and it seemed to only render the nodeName when printed in the console. Thought it was better practice to stay consistent, but can hardcode it as well if that's what we want to do. I could also be explicit and do ${inputRef.current.nodeName}

Reply:
Let's have it hardcoded here. There is no point in having a variable here and the code will be a tiny bit smaller and run a tiny bit faster.